redwood/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+#![deny(clippy::all)]
+
+use pyo3::create_exception;
+use pyo3::exceptions::PyException;
+use pyo3::prelude::*;
+use sequoia_openpgp::cert::{CertBuilder, CipherSuite};
+use sequoia_openpgp::crypto::Password;
+use sequoia_openpgp::parse::{stream::DecryptorBuilder, Parse};
+use sequoia_openpgp::policy::StandardPolicy;
+use sequoia_openpgp::serialize::{
+    stream::{Armorer, Encryptor, LiteralWriter, Message},
+    SerializeInto,
+};
+use sequoia_openpgp::Cert;
+use std::fs::File;
+use std::io::{self, Read};
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+use std::string::FromUtf8Error;
+use std::time::{Duration, SystemTime};
+
+/// Alias to make it easier for Python readers
+type Bytes = Vec<u8>;
+
+mod decryption;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("OpenPGP error: {0}")]
+    OpenPgp(#[from] anyhow::Error),
+    #[error("IO error: {0}")]
+    Io(#[from] io::Error),
+    #[error("Unexpected non-UTF-8 text: {0}")]
+    NotUnicode(#[from] FromUtf8Error),
+}
+
+create_exception!(redwood, RedwoodError, PyException);
+
+impl From<Error> for PyErr {
+    fn from(original: Error) -> Self {
+        // TODO: make sure we're not losing the stacktrace here
```

Comment:
I think the answer is that we're not losing the stacktrace because it never had it in the first place :/ https://docs.rs/anyhow/latest/anyhow/ describes that we need to enable both the `backtrace` feature and set an environment variable.

Based on https://docs.rs/anyhow/latest/anyhow/struct.Error.html#display-representations we should at least use the Debug mode: `{:?}` so we're not dropping context

Reply:
pyo3 now supports a built-in conversion from anyhow (https://github.com/PyO3/pyo3/blob/342e4ca678834736779e6ada4d60354be7bd4b1d/src/conversions/anyhow.rs#L114), I'm trying to see if that's enough to drop our `Error` type entirely and just go from anyhow::Error -> RedwoodError...